aten/src/ATen/native/EmbeddingBag.cpp

```diff
@@ -86,11 +86,11 @@ void index_select_add<float>(const Tensor &select_indices,
                              const Tensor& offsets,
                              bool include_last_offset) {
   int64_t ddim = src.size(1);
-  auto* src_data = src.data_ptr<float>();
   auto* select_indices_data = select_indices.data_ptr<int64_t>();
   auto* output_data = output.data_ptr<float>();
 
   if (isFastPathIndexSelect(src, output)) {
+    auto* src_data = src.contiguous().data_ptr<float>();
```

Comment:
wait, this is still a bug, right? If the tensor is non-contiguous then `contiguous()` will return a new tensor and it will be immediately destroyed (because we don't keep a reference to it around). So `src_data` will point to the deallocated memory :(

I wonder why ASAN doesn't catch it.

It should be
```
auto src_contig = src.contiguous();
auto* src_data = src_contig.data_ptr<float>();
```

cc @glaringlee @ngimel

Reply:
@dzhulgakov oh, shoot.........my bad, will fix soon.